### PR TITLE
Implement lambda shortener

### PR DESCRIPTION
**Pull request by Nekotekina**
Something similar existed earlier in RPCS3 repo but kinda different.

### CODE REVIEW

File: rpcs3/util/types.hpp [hunk 1]
```diff
@@ -55,6 +55,48 @@ using namespace std::literals;
 #define AUDIT(...) (static_cast<void>(0))
 #endif
 
+namespace utils
+{
+	template <typename F>
+	struct fu_helper
```
Comment (Megamouse) [inline rpcs3/util/types.hpp:61]: why not use fun helper and FUN

Comment (Nekotekina) [reply rpcs3/util/types.hpp:61]: is FU so bad?

Comment (MSuih) [reply rpcs3/util/types.hpp:61]: I would prefer fun helper over fuck you helper. Positivity is better and I don't think one additional letter is too big of a burden.

Comment (Nekotekina) [reply rpcs3/util/types.hpp:61]: Okay

File: rpcs3/main.cpp [hunk 2]
```diff
@@ -473,7 +473,7 @@ int main(int argc, char** argv)
 	WSAStartup(MAKEWORD(2, 2), &wsa_data);
 #endif
 
-	ensure(thread_ctrl::is_main(), "Not main thread");
+	ensure(thread_ctrl::is_main(), FN(x), "Not main thread");
```
Comment (Megamouse) [inline rpcs3/main.cpp:476]: i have to be honest. I have no idea what this does.

Comment (Nekotekina) [reply rpcs3/main.cpp:476]: Nothing, i.e. it takes first argument and returns it.

Comment (Megamouse) [reply rpcs3/main.cpp:476]: but why is it needed

Comment (Megamouse) [reply rpcs3/main.cpp:476]: you have an overload if i'm not mistaken.
So it seems kinda useless

Comment (Nekotekina) [reply rpcs3/main.cpp:476]: Right, it can be solved with overload resolution.

**Pull request merged**